clippy_lints/src/unit_types/let_unit_value.rs

```diff
@@ -13,56 +13,68 @@ use rustc_middle::ty;
 use super::LET_UNIT_VALUE;
 
 pub(super) fn check<'tcx>(cx: &LateContext<'tcx>, local: &'tcx Local<'_>) {
-    if let Some(init) = local.init
-        && !local.pat.span.from_expansion()
-        && !in_external_macro(cx.sess(), local.span)
-        && cx.typeck_results().pat_ty(local.pat).is_unit()
-    {
-        if (local.ty.map_or(false, |ty| !matches!(ty.kind, TyKind::Infer))
-            || matches!(local.pat.kind, PatKind::Tuple([], ddpos) if ddpos.as_opt_usize().is_none()))
-            && expr_needs_inferred_result(cx, init)
-        {
-            if !matches!(local.pat.kind, PatKind::Wild)
-               && !matches!(local.pat.kind, PatKind::Tuple([], ddpos) if ddpos.as_opt_usize().is_none())
+    // skip `let () = { ... }`
+    if let PatKind::Tuple(fields, ..) = local.pat.kind && fields.is_empty() {
+        return;
+    }
+
+    if_chain! {
+        if let Some(init) = local.init;
+        if !local.pat.span.from_expansion();
+        if !in_external_macro(cx.sess(), local.span);
+        if cx.typeck_results().pat_ty(local.pat).is_unit();
+        then {
```

Comment:
```suggestion
        if let Some(init) = local.init;
        if !local.pat.span.from_expansion() &&
            !in_external_macro(cx.sess(), local.span) &&
            cx.typeck_results().pat_ty(local.pat).is_unit() {
```

Also remove the stray `}` on the way out.

Reply:
I found it a tad easier to read in this case, but I'm ok with switching it back.

Reply:
Fixed

Reply:
I've just seen this zulip discussion - https://rust-lang.zulipchat.com/#narrow/stream/257328-clippy/topic/removing.20if_chain.20dependency/near/292061902, didn't know clippy will at some point transition away from them, so I'll make sure to use let chains instead in future PRs